server/controllers/profile.js

```diff
@@ -1,77 +1,87 @@
-const mongoose = require("mongoose");
+const mongoose = require('mongoose');
 
-const asyncHandler = require("express-async-handler");
+const asyncHandler = require('express-async-handler');
 
-const Profile = require("../models/Profile");
+const Profile = require('../models/Profile');
 
 // @route POST /profile
 // Add a new profile
-exports.createProfile = asyncHandler(async (req, res, next) => {
-  const profile = req.body;
-  const newProfile = new Profile({ ...profile });
+exports.createProfile = asyncHandler(async(req, res, next) => {
+    const profile = req.body;
+    const newProfile = new Profile({...profile });
 
-  if (!newProfile) {
-    res.status(500);
-    throw new Error("Profile could not be created.");
-  }
-  await newProfile.save();
-  res.status(201).json({ profile: newProfile });
+    if (!newProfile) {
+        res.status(500);
+        throw new Error('Profile could not be created.');
+    }
+    await newProfile.save();
+    res.status(201).json({ profile: newProfile });
 });
 
 // @route GET /profile
 // Get all profiles
-exports.allProfiles = asyncHandler(async (req, res, next) => {
-  const profiles = await Profile.find();
-  if (!profiles) {
-    res.status(404);
-    throw new Error("Cannot access server.");
-  }
-  res.status(200).json({ profiles });
+exports.allProfiles = asyncHandler(async(req, res, next) => {
+    const profiles = await Profile.find();
+    if (!profiles) {
+        res.status(404);
+        throw new Error('Cannot access server.');
```

Comment:
If there are no profiles, it should return an empty array, you can return it, rather Missing Resource